crates/adapters/celestia/src/da_service/mod.rs

```diff
@@ -192,6 +199,20 @@ impl CelestiaService {
             );
         }
 
+        let tx_priority = config.tx_priority.clone().into();
+        if let Ok(signer) = client.address() {
+            let bg_client = config
+                .build_client()
+                .await
+                .expect("Failed to build celestia-client for background task");
+            tokio::spawn(stat_collection_task(
```

Comment:
Shouldn't we register the join handle of this long-running task?

Reply:
I thought about it, but we don't need to wait for this task on restart, because it does not interact with the rest of the system, only metrics. 

So the worst case scenario, there might be 2 metrics around restart, and only in test.